### PR TITLE
✨ guild settings command

### LINKED ISSUE #55: /settings allows overriding other user's settings

### PR DESCRIPTION
**Pull request by dudeofawesome**
- Adds the Guild settings command
- Upgrades Discord NestJS
- Adds permissions validation to settings command (required upgrading Discord NestJS)
- Enables commands for all users
- Disables commands in DMs

Fixes #55

### CODE REVIEW

File: src/bot/commands/guild-settings.command.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  @Length(1, 1, { message: '`role_emoji` must be a single character' })
+  @Matches(/^\p{Extended_Pictographic}$/u, {
+    message: '`role_emoji` must be an emoji',
+  })
+  role_emoji?: string;
+}
+
+@Command({
+  name: 'guild-settings',
+  description: `Changes your guild's settings for Virginity Bot`,
```
Comment (EdgarSaldivar) [inline src/bot/commands/guild-settings.command.ts]: add a '.' at the end, for good grammar.

File: src/bot/commands/guild-settings.command.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  role_color?: HexColorString;
+
+  /** An emoji to adorn the biggest virgin's emoji. */
+  @Param({
+    description: `An emoji to adorn the biggest virgin's emoji.`,
+    required: false,
+    type: ParamType.STRING,
+  })
+  @IsOptional()
+  @Length(1, 1, { message: '`role_emoji` must be a single character' })
```
Comment (EdgarSaldivar) [inline src/bot/commands/guild-settings.command.ts:151]: `.` here too?

File: src/bot/commands/guild-settings.command.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      dto.channel_description ?? guild_ent.channel.description;
+
+    guild_ent.role.name = dto.role_name ?? guild_ent.role.name;
+    guild_ent.role.color = dto.role_color ?? guild_ent.role.color;
+    guild_ent.role.emoji = dto.role_emoji ?? guild_ent.role.emoji;
+
+    await this.guilds.flush();
+
+    return new MessagePayload(interaction.channel, {
+      content: 'Updated guild settings',
```
Comment (EdgarSaldivar) [inline src/bot/commands/guild-settings.command.ts]: Add a period '.'?

**Review by EdgarSaldivar — COMMENTED** [review]
10xer 4sure

File: src/bot/commands/guild-settings.command.ts (hunk 2 above)
Comment (dudeofawesome) [reply src/bot/commands/guild-settings.command.ts:151]: So this actually shouldn't have a period because the `.` gets added later.

**Pull request merged**